Sources/NIO/EventLoop.swift

```diff
@@ -396,7 +515,10 @@ internal final class SelectableEventLoop: EventLoop {
             promise.fail(error: error)
         },`in`)
 
-        let scheduled = Scheduled(promise: promise, cancellationTask: {
+        let scheduled = Scheduled(promise: promise, cancellationTask: { [weak self] in
+            guard let `self` = self else {
```

Comment:
we don't normally use weak references because their impact on performance. Why is this one necessary?